src/server/stream.rs

```diff
@@ -229,13 +229,16 @@ impl<S: SecuritySwitch> Write for SwitchingTlsStream<S> {
     }
 
     fn flush(&mut self) -> io::Result<()> {
+        debug!("Flush called <<{}>>", self.channel);
         let state = self.state.lock().unwrap().which_state(self.channel);
         match state {
             SecurityState::On => {
                 self.tls.flush()?;
                 while self.tls.wants_write() {
                     self.tls_io(TlsIoType::Write)?;
                 }
+                self.tls.flush()?;
```

Comment:
Do we have to flush tls twice?

Reply:
I'm not sure to be honest and not sure how to be sure. Its precautionary.

Reply:
Perhaps we don't need the first flush

Reply:
I've removed it for now.